src/N3DataFactory.js

```diff
@@ -256,6 +265,53 @@ export class Quad {
 }
 export { Quad as Triple };
 
+// QuadTerm constructor
+export class QuadTerm extends Term {
+  constructor(quad) {
+    super(`<<${quadToId(quad)}>>`);
+    this.quad = quad;
+  }
+
+  // ### The term type of this term
+  get termType() {
+    return 'QuadTerm';
+  }
+
+  // ### The text value of this literal
+  get value() {
+    return this.quad;
+  }
+}
+
+// ### Creates a string from a Quad
+function quadToId(quad) {
+  let result = [quad.subject, quad.predicate, quad.object].map(termToId).map(escape).join(' ');
+
+  // Omit default graph
+  if (!isDefaultGraph(quad.graph)) {
+    result += ` ${termToId(quad.graph)}`;
+  }
+  return result;
+}
+
+// ### Escapes the quotes within the given literal
+function escape(id) {
+  return escapeHelper(id, /"/g, '""');
+}
+
+// ### Unescapes the quotes within the given literal
+function unEscape(id) {
+  return escapeHelper(id, /""/g, '"');
+}
+
+function escapeHelper(id, regex, replacement) {
```

Comment:
How about simply (warning, untested):

```javascript
return id.replace(/^"(.*".*)(?="[^"]*$)/, (_, quoted) => `"${quoted.replace(regex, replacement)}`);
```

Note how the regex requires the presence of a quote before substitution if even considered (and the tail replacement is a non-captured lookahead).

Reply:
I don't like the name `escapeHelper`, but if it is indeed possible to make it a one-liner as above, we could just simply put the first regex in a constant, and then inline the replacements in the `escape` and `unescape` functions.